### PR TITLE
Add song title and artist name to tray menu

### PR DESCRIPTION
**Pull request by nick41496**
Added a menu item to the tray menu to display the song title and artist name for quick info without having to switch windows/virtual desktops.

### CODE REVIEW

File: src/main/features/core/tray.js [hunk 1]
```diff
@@ -29,6 +29,18 @@ function getAppIconFileName() {
   return path.resolve(currentIconPath, appIconFileName);
 }
 
+// Trim the song title to fit in the menu
+function getSongListing(song) {
```
Comment (MarshallOfSound) [inline src/main/features/core/tray.js]: ES6 ify this please

```js
const getTrackString = (song) => {
  // Do things
};
```

File: src/main/features/core/tray.js [hunk 2]
```diff
@@ -29,6 +29,18 @@ function getAppIconFileName() {
   return path.resolve(currentIconPath, appIconFileName);
 }
 
+// Trim the song title to fit in the menu
```
Comment (MarshallOfSound) [inline src/main/features/core/tray.js]: Although in general comments are nice, if it's obvious what the function does they just clutter the code IMO

File: src/main/features/core/tray.js [hunk 3]
```diff
@@ -61,14 +73,17 @@ Emitter.on('playback:isStopped', () => {
   appIcon.setImage(getAppIconFileName());
 });
 
-const setContextMenu = () => {
+const setContextMenu = (song) => {
```
Comment (MarshallOfSound) [inline src/main/features/core/tray.js]: Since `4.0.0` we have been referring to the current playing track as "track" internally to match up with other libraries we use and for general consistency

s/song/track

File: src/main/features/core/tray.js [hunk 4]
```diff
@@ -61,14 +73,17 @@ Emitter.on('playback:isStopped', () => {
...
   const contextMenu = Menu.buildFromTemplate([
     { label: TranslationProvider.query('tray-label-show'),
       click: () => {
         mainWindow.setSkipTaskbar(false);
         mainWindow.show();
       },
     },
+    { label: song ? getSongListing(song) : 'Not playing',
```
Comment (MarshallOfSound) [inline src/main/features/core/tray.js]: "Not Playing" must be a translation string

File: src/main/features/core/tray.js [hunk 5]
```diff
@@ -219,3 +234,8 @@ Emitter.on('audiooutput:list', (event, devices) => {
```
Comment (MarshallOfSound) [inline src/main/features/core/tray.js]: I can't actually comment on the line I want to, but line 232

```js
setContextMenu();
```

will cause your logic to break.  You aren't passing in a track so it will reset the string to "Not Playing".  You should pass in `PlaybackAPI.currentSong()` in this call.

**Sidenote:** I need to update `currentSong` to be `currentTrack` 😆

File: src/main/features/core/tray.js [hunk 6]
```diff
@@ -219,3 +234,8 @@ Emitter.on('audiooutput:list', (event, devices) => {
 
 Emitter.sendToGooglePlayMusic('audiooutput:fetch');
 Emitter.on('audiooutput:set', () => Emitter.sendToGooglePlayMusic('audiooutput:fetch'));
+
+// When the track changes reset the context menu to reflect the track change
```
Comment (MarshallOfSound) [inline src/main/features/core/tray.js]: Same thing about comments 👍

**Comment by nick41496** [comment]
menu_001
menu_002
menu_003

File: src/_locales/de.json [hunk 7]
```diff
@@ -74,7 +74,7 @@
   "settings-option-custom-theme-custom-color": "eigene Farbe",
   "settings-option-custom-theme-dark": "Dunkel",
   "settings-option-custom-theme-light": "Hell",
-  "settings-option-custom-theme-should-track-system": "",
+  "settings-option-custom-theme-should-track-system": "Farbschema an Dark-Mode-Option des Systems anpassen",
```
Comment (jostrander) [inline src/_locales/de.json]: This should be in it's own PR imo.

**Comment by MarshallOfSound** [comment]
Thanks @nick41496 looking good 👍
